modules/solid_mechanics/include/neml2/utils/NEML2Utils.h

```diff
@@ -43,6 +43,12 @@ namespace NEML2Utils
 
 #ifdef NEML2_ENABLED
 
+/// special version of get_model that adds a MooseApp parameter
+neml2::Model & get_model_moose(const std::string & mname,
+                               MooseBase * moose_base,
+                               bool enable_ad = true,
+                               bool force_create = true);
```

Comment:
What's the reason for not directly passing a pointer to MooseApp?

Reply:
Because with this signature the caller can just pas `this` into `get_model_moose`.